### DISCUSSION TITLE
Detect if an element loses focus

### DISCUSSION THREAD

**Discussion by dfallman** [alpinejs/alpine#2411] [1. Help] [answered]
I'm looking to find a way to use Alpine to detect if an element _loses_ focus.

I'm using the `x-on:focus="..."` to set a variable to true when (in this case) an input field has focus, but is there a way to do the reverse with alpine -- something like `x-on:focus.away="..."` (that was my first guess, but doesn't work!)

The code looks something like:

```
<input type="text" x-ref="ref" name="title" x-model="title" class="..." 
            @focus="doStuff=true" 
    >>> @focus.away="doStuff=false" <<< 
>
```

**Comment by danddanddand** [comment, marked as answer]
https://developer.mozilla.org/en-US/docs/Web/API/Element/blur_event

**Reply by dfallman** [reply]
Perfect, thanks -- wish all the things you can do with `x-on` were listed/documented somewhere

**Reply by ekwoka** [reply]
@dfallman They are, on MDN, in the Element `events` list.

**Comment by charliex311** [comment]
For your case, `x-on:blur` must be working fine.
